docs/why.qmd

```diff
@@ -2,24 +2,224 @@
-work with data in multiple data platforms, find yourself translating between
-other Python dataframe APIs, or just want a great Python dataframe experience,
-Ibis is for you.
+We can demonstrate this with a simple example on a few local query engines:
+
+```{python}
+import ibis
+
+ibis.options.interactive = True
+```
+
+::: {.panel-tabset}
+
+```{python}
+#| echo: false
+t = ibis.examples.penguins.fetch()
+t.to_parquet("penguins.parquet")
+```
+
+## DuckDB
+
+```{python}
+con = ibis.duckdb.connect()  # <1>
+
+t = con.read_parquet("penguins.parquet")
+t.limit(3)
+```
+
+1. Change only your connection to switch between backends.
+
+```{python}
+t.group_by(["species", "island"]).agg(count=t.count()).order_by("count")
+```
+
+## Polars
+
+```{python}
+con = ibis.polars.connect()  # <1>
+
+t = con.read_parquet("penguins.parquet")
+t.limit(3)
+```
+
+1. Change only your connection to switch between backends.
+
+```{python}
+t.group_by(["species", "island"]).agg(count=t.count()).order_by("count")
+```
+
+## DataFusion
+
+```{python}
+con = ibis.datafusion.connect()  # <1>
+
+t = con.read_parquet("penguins.parquet")
+t.limit(3)
+```
+
+1. Change only your connection to switch between backends.
+
+```{python}
+t.group_by(["species", "island"]).agg(count=t.count()).order_by("count")
+```
+
+## PySpark
+
+```{python}
+con = ibis.connect("pyspark://")  # <1>
+
+t = con.read_parquet("penguins.parquet")
+t.limit(3)
+```
+
+1. Change only your connection to switch between backends.
+
+```{python}
+t.group_by(["species", "island"]).agg(count=t.count()).order_by("count")
+```
+
+:::
+
+## Who is Ibis for?
+
+Ibis is for data engineers, data analysts, and data scientists to use directly
+with their data platform(s) of choice. It also has benefits for [data
+platforms](#ibis-for-data-platforms), [data
+organizations](#ibis-for-data-organizations), and [data library
+developers](#ibis-for-data-library-developers).
+
+### Ibis for data practitioners
 
-## Portability
+You can use Ibis at any stage of your data workflow, no matter your role.
 
-You can reuse the same code across different backends.
+**Data engineers** can use Ibis to:
+
+- write and maintain complex ETL/ELT jobs
+- replace fragile SQL string pipelines with a robust Python API
+- replace PySpark with a more Pythonic API that supports Spark and many other
+ backends
+
+**Data analysts** can use Ibis to:
+
+- use Ibis interactive mode for rapid exploration
+- perform rapid exploratory data analysis using interactive mode
+- [create end-to-end analytics workflows](./posts/ibis-analytics/index.qmd)
+- work in a general-purpose, yet easy to learn, programming language without the
+ need for formatting SQL strings
+
+**Data scientists** can use Ibis to:
+
+- extract a sample of data for local iteration with a fast local backend
+- prototype with the same API that will be used in production
+- preprocess and feature engineer data before training a machine learning model
+
+### Ibis for data platforms
```

Comment:
i personally think data platforms/engines are integration partners, not users. For that reason, I almost feel like we should have a separate blog to focus on the whys for partners, to keep this one focused and avoid any potential confusion to users.

Reply:
I'll leave this as a follow up blog -- "Ibis for data platforms" -- I think this still makes sense here